src/webots/core/WbProject.cpp

```diff
@@ -250,12 +250,7 @@ bool WbProject::createNewProjectFolders() {
 }
 
 bool WbProject::isReadOnly() const {
-#ifdef _WIN32
-  Qt::CaseSensitivity caseSensitivity = Qt::CaseInsensitive;
-#else
-  Qt::CaseSensitivity caseSensitivity = Qt::CaseSensitive;
-#endif
-  return mPath.startsWith(WbStandardPaths::webotsHomePath(), caseSensitivity);
+  return WbFileUtil::isLocatedInDirectory(mPath, WbStandardPaths::webotsHomePath());
```

Comment:
```suggestion
  return WbFileUtil::isLocatedInInstallationDirectory(mPath);
```
To generalize this replacement, we should probably add a boolean argument to WbFileUtil::isLocatedInInstallationDirectory to ignore the test on `WEBOTS_ALLOW_MODIFY_INSTALLATION` if not wanted, but in this case, I believe we can call this method as is.

Reply:
I'm not sure it's a good idea to leave it as-is, as it might vary the behavior significantly. All it takes is for you to have, at any given time, installed the development environment and this variable will exist in your bashrc. This specific case is used (I think) for example to determined the behavior when you right-click a PROTO and which options should be there (view and/or edit). We can do the change you mention (calling isLocatedInInstallationDirectory instead of isLocatedInDirectory), but it feels like altering the logic (i.e., checking also for WEBOTS_ALLOW_MODIFY_INSTALLATION) now is a bit risky